sessionctx/variable/tidb_vars.go

```diff
@@ -771,6 +771,9 @@ const (
 	// limit for ranges.
 	TiDBOptRangeMaxSize = "tidb_opt_range_max_size"
 
+	// TiDBOptAdvancedJoinHint indicates whether the join method hint is compatible with join order hint.
+	TiDBOptAdvancedJoinHint = "tidb_opt_advanced_join_hint"
```

Comment:
Have we discussed the naming with the PM?
This is an abstract name and doesn't indicate the concrete effect of this variable. Personally, I don't think it's a good name.